zerver/migrations/0479_backfill_edit_history.py

```diff
@@ -0,0 +1,47 @@
+# Generated by Django 3.2.2 on 2021-05-11 10:00
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import JSONField
+from django.db.models.functions import Cast
+
+BATCH_SIZE = 5000
+
+
+def backfill_edit_history(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
```

Comment:
The way this is deployed in Zulip Cloud, migrations run on staging before prod gets the new code.  So if the backfill migration is in the same commit as the double-writes, then we backfill the old entries, but prod will still be updating `edit_history` without updating `edit_history_entries` in the window between when the backfill ends and when prod gets the double-writing code.

This is also necessary because more message rows will be being written as the migration runs, and it only looks at the latest message `id` at the start.  So we need a guarantee that all of the _new_ rows that are written during this migration are double-writes, which means the double-writes must be on prod before the backfill is started.

So I think the first deploy point is 0478 (to add the column) with the double-write code, but without the backfill migration? We can do that by manually by fudging the migration in production, but it seems like it'd be reasonably easy to split this commit to accomplish that.

There _is_ still a race here where staging is updated with the double-write code, and prod isn't -- staging could double-write both `edit_history` and `edit_history_entries`, skipping it from the later backfill, and prod could then update only `edit_history`, causing them to be out of sync.  But staging is locked down, so we can just ... not do that. :)

Reply:
> There is still a race here where staging is updated with the double-write code, and prod isn't -- staging could double-write both `edit_history` and `edit_history_entries`, skipping it from the later backfill, and prod could then update only `edit_history`, causing them to be out of sync. But staging is locked down, so we can just ... not do that. :)

We could skip skipping the presumably double-written rows in the backfilling migration, if we really want to eliminate this race.

Reply:
Sure.

Reply:
Ah, I missed that.  Pulling it out into its own commit is appreciated.